src/general/VersionUtils.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Helpers for dealing with Thrive's version number
+/// </summary>
+public static class VersionUtils
+{
+    public static readonly string[] Suffixes =
+    {
+        "PRE-ALPHA",
+        "ALPHA",
+        "BETA",
+        "RC1",
+        "RC2",
+        "RC3",
+        "RC4",
+        "RC5",
+        "RC6",
+        "RC7",
+        "RC8",
+        "RC9",
+    };
+
+    /// <summary>
+    ///   Compare the given version numbers.
+    /// </summary>
+    /// <returns>
+    ///   0 if the versions are the same,
+    ///   a negative integer if a is a smaller (older) version than b and
+    ///   a positive integer if a is a bigger (newer) version than b
+    /// </returns>
+    /// <param name="a">The first version to compare.</param>
+    /// <param name="b">The second version to compare.</param>
+    public static int Compare(string a, string b)
+    {
+        if (a == b)
+        {
+            return 0;
+        }
+
+        char[] separator = { '-' };
+        var aSplit = a.Split(separator, 2);
+        var bSplit = b.Split(separator, 2);
+
+        // Compare the numeric versions
+        try
+        {
+            int versionDiff = Version.Parse(aSplit[0]).CompareTo(Version.Parse(bSplit[0]));
+            if (versionDiff != 0)
+            {
+                return versionDiff;
+            }
+        }
+        catch (Exception e)
+        {
+            GD.PrintErr(e);
+        }
+
+        // If only one of the version has a suffix, that one is older
+        if (aSplit.Length != bSplit.Length)
+        {
+            return bSplit.Length - aSplit.Length;
+        }
+
+        // Compare predefined suffixes
+        int aSuffixIndex = Array.IndexOf(Suffixes, aSplit[1].ToUpperInvariant());
```

Comment:
With the constant case change I suppose `ToLowerInvariant` would be needed here?

Reply:
Interesting. Wouldn't then just ToLower work just fine with `ToLower(CultureInfo.InvariantCulture)`?

Reply:
That also told me to use ToUpperInvariant.

Reply:
Could you show me the warning / error message?

Reply:
VersionUtils.cs(52,52): Warning CA1308: In method 'Compare', replace the call to 'ToLowerInvariant' with 'ToUpperInvariant'. (CA1308) (Thrive)

Reply:
I think we should just globally ignore that warning as we don't care about round tripping any data.
Ignoring it should work by adding that rule here: https://github.com/Revolutionary-Games/Thrive/blob/master/StyleCop.ruleset#L22 and setting it to none.